src/main/java/tc/oc/pgm/modules/MaxBuildHeightMatchModule.java

```diff
@@ -33,4 +34,10 @@ public void checkBuildHeight(BlockTransformEvent event) {
       }
     }
   }
+
+  private int getExtraHeight(Material material) {
+    if (material.name().contains("DOOR") && !material.name().contains("TRAP")) return 1;
```

Comment:
Can we put this in a util? Also, `BlockTransformEvent` is a very, *very* hot part of the code: Can we optimize this with enums and switches? 

I would even check have two checks, so this will be short-circuited.
```java
if (event.getNewState().getY() < this.buildHeight - 1) return;
if (event.getNewState().getY()
    + getExtraHeight(event.getNewState().getMaterial()) < this.buildHeight) return;
// setCancelled

Reply:
Also, this isn't just an issue with `MaxBuildHeightMatchModule`, right? The bug will still happen for `RegionMatchModule`, can we fix it for both?

Reply:
Code changed in latest commit, re-review